### PR TITLE
Feature pl2 asset manager

### PR DESCRIPTION
**Pull request by dknuth0101**

### CODE REVIEW

**Comment by dknuth0101** [comment]
This crap doesn't work

File: d2common/d2data/d2datadict/palette_transform.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package d2datadict
+
+import (
+	"github.com/OpenDiablo2/OpenDiablo2/d2common/d2fileformats/d2pl2"
+	"github.com/OpenDiablo2/OpenDiablo2/d2common/d2enum"
+)
+
+// PaletteTransformType represents a paletteTransform
+type PaletteTransformRec struct {
+	Name	d2enum.PaletteTransformType
```
Comment (FooSoft) [inline d2common/d2data/d2datadict/palette_transform.go]: Don't need `PaletteTransformType`

File: d2common/d2data/d2datadict/palette_transform.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	Data	d2pl2.PL2File
+}
+
+var PaletteTransforms map[d2enum.PaletteTransformType]PaletteTransformRec
```
Comment (FooSoft) [inline d2common/d2data/d2datadict/palette_transform.go]: Don't need

File: d2common/d2data/d2datadict/palette_transform.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+// CreatePaletteTransform creates a paletteTransform transform
+func CreatePaletteTransform(name d2enum.PaletteTransformType, data []byte) PaletteTransformRec {
```
Comment (FooSoft) [inline d2common/d2data/d2datadict/palette_transform.go]: Only need `[]byte`

File: d2common/d2data/d2datadict/palette_transform.go (hunk 1 above)
Comment (FooSoft) [reply d2common/d2data/d2datadict/palette_transform.go]: You don't need to load anything other than `d2pl2.PL2File`

File: d2common/d2enum/palette_transform_defs.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
+package d2enum
+
+// PaletteTransformType represents a named paletteTransform
+type PaletteTransformType string
+
+const (
```
Comment (FooSoft) [inline d2common/d2enum/palette_transform_defs.go]: Don't need these constants

File: d2common/d2fileformats/d2pl2/pl2.go [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+// uses restruct to read the binary dc6 data into structs
+func LoadPL2(data []byte) (PL2File, error) {
+	result := &PL2File{}
+
+	restruct.EnableExprBeta()
+	err := restruct.Unpack(data, binary.LittleEndian, &result)
+	if err != nil {
+		log.Printf("failed to read pl2: %v", err)
```
Comment (FooSoft) [inline d2common/d2fileformats/d2pl2/pl2.go]: Don't print

File: main.go [hunk 6]
```diff
@@ -180,6 +180,10 @@ func initialize() error {
 		return err
 	}
 
+	if err := loadPaletteTransforms(); err != nil {
```
Comment (FooSoft) [inline main.go]: Don't need to do this, load on demand

File: main.go [hunk 7]
```diff
@@ -402,6 +406,39 @@ func loadPalettes() error {
 	return nil
 }
 
+func loadPaletteTransforms() error {
```
Comment (FooSoft) [inline main.go]: Don't need this, callers can just provide palette path instead of some constant

**Pull request closed without merging**